File: benchmarks/runtime/regex_match.lua
local n = 50000
local i = 0
local sum = 0
while i < n do
  local ok = string.find("hello-123", "[a-z]+%-[0-9]+")
  if ok then sum = sum + 1 end
  local s = string.gsub("abc-123-xyz", "[0-9]+", "N")
  sum = sum + #s
  i = i + 1
end
if sum <= 0 then os.exit(1) end
print(sum)

File: benchmarks/runtime/regex_match.js
let n = 50000;
let i = 0;
let sum = 0;
const pat = /[a-z]+-[0-9]+/;
while (i < n) {
  const ok = pat.test("hello-123") ? 1 : 0;
  sum += ok;
  const s = "abc-123-xyz".replace(/[0-9]+/g, "N");
  sum += s.length;
  i += 1;
}
if (sum <= 0) process.exit(1);
console.log(sum);
